app/controllers/teacher_interface/uploads_controller.rb

```diff
@@ -25,15 +25,23 @@ def create
     end
 
     def delete
+      @delete_file_form = DeleteFileForm.new
     end
 
     def destroy
-      if ActiveModel::Type::Boolean.new.cast(params.dig(:upload, :confirm))
-        @upload.attachment.purge
-        @upload.destroy!
+      confirm = params.dig(:teacher_interface_delete_file_form, :confirm)
+      @delete_file_form = TeacherInterface::DeleteFileForm.new(confirm:)
+      if @delete_file_form.valid?
+        if @delete_file_form.confirm
+          @upload.attachment.purge
+          @upload.destroy!
+          redirect_to [:edit, :teacher_interface, :application_form, @document]
+        elsif @delete_file_form.confirm == false
+          redirect_to [:edit, :teacher_interface, :application_form, @document]
+        end
```

Comment:
```suggestion
        end
        redirect_to [:edit, :teacher_interface, :application_form, @document]
```

Since both branches of the if statement do the same thing at the end.